packages/react-charting/src/components/HorizontalBarChart/HorizontalBarChart.base.tsx

```diff
@@ -286,19 +295,47 @@ export class HorizontalBarChartBase extends React.Component<IHorizontalBarChartP
     );
   }
 
+  // Adjusts the margin percentage, based on the width of the svg (inversly propotional)
+  private _adjustBarSpacing(): void {
+    const svgWidth = this.barChartSvgRef.current?.getBoundingClientRect().width;
+    if (svgWidth) {
+      // [0,1] Controls the spacing between the bars
+      const BAR_SPACING_FACTOR = 1;
+      // The barspacing is provided in percentage terms, so changes inversely to the width of the svg
+      const currentBarSpacing = clamp((200 / svgWidth) * BAR_SPACING_FACTOR, 1, 0.2);
+      this.setState({ barSpacing: currentBarSpacing });
+    }
+  }
+
+  /**
+   * This functions returns an array of <rect> elements, which form the bars
+   * For each bar an x value, and a width needs to be specified
+   * The computations are done based on percentages
+   * Extra margin is also provided, in the x value to provide some spacing in between the bars
+   */
+
   private _createBars(data: IChartProps, palette: IPalette): JSX.Element[] {
+    const noOfBars =
+      data.chartData?.reduce(
+        (count: number, point: IChartDataPoint) => (count += (point.horizontalBarChartdata?.x || 0) > 0 ? 1 : 0),
+        0,
+      ) || 1;
+    const totalSpacing = this.state.barSpacing * (noOfBars - 1);
```

Comment:
This barSpacing is set after the svgWidth can be accessed (after the component mounts). So once barSpacing is set we need to rerender the component to relay the change to the component, else it will render broken state after the first render until a rerender is triggered (due to some other event, like hover, click, etc).

Hence created it as a state